src/transformers/generation/continuous_batching/continuous_api.py

```diff
@@ -1147,42 +1167,40 @@ def generate_batch(
             progress_bar = False
 
         # Initialize manager with the batch inputs
-        manager = self.init_continuous_batching(
+        with self.continuous_batching_context_manager(
             generation_config=generation_config,
             num_q_cuda_graphs=num_q_cuda_graphs,
             num_kv_cuda_graphs=num_kv_cuda_graphs,
-        )
-        manager.start()
-        results = {}
-        num_requests = len(inputs)
-        try:
-            from tqdm.contrib.logging import logging_redirect_tqdm
-
-            with logging_redirect_tqdm([logger]):
-                with tqdm(
-                    total=num_requests,
-                    disable=(not progress_bar),
-                    desc=f"Solving {num_requests} requests",
-                    unit="request",
-                ) as pbar:
-                    manager.add_requests(inputs=inputs, max_new_tokens=kwargs.get("max_new_tokens"))
-                    finished_count = 0
-                    while finished_count < num_requests:
-                        result = manager.get_result(timeout=1)
-                        if result:
-                            req_id = result.request_id
-                            if result.is_finished():
-                                results[req_id] = result
-                                finished_count += 1
-                                pbar.update(1)
-                        else:
-                            if not manager.is_running():
-                                logger.error("Generation thread terminated unexpectedly.")
-                                break
+            block=True,
+            timeout=5,
+        ) as manager:
+            results = {}
+            num_requests = len(inputs)
+            try:
+                from tqdm.contrib.logging import logging_redirect_tqdm
+
+                with logging_redirect_tqdm([logger]):
```

Comment:
to decrease indentation level, we put the two context managers on the same line

Reply:
`>:(`